packages/sdk-js/src/GrowthBook.ts

```diff
@@ -1483,7 +1513,8 @@ export class GrowthBook<
     }
 
     // 14. Fire the tracking callback
-    this._track(experiment, result);
+    // Store the promise in case we're awaiting it (ex: browser url redirects)
+    this._activeTrackingCall = this._track(experiment, result);
```

Comment:
I don't like the idea of storing this globally.  Seems like it might open us up to race conditions.  Instead, we could include this promise in the return value of the function.

Reply:
pinged you, but I kind of disagree, unless I'm misunderstanding your suggestion:
- We'd need to add something like `result.trackingCall` which may not serialize well for people consuming/logging `result` directly. We might break existing integrations.
- the global `_activeTrackingCall` is set and consumed in the same exe cycle since `_runAutoExperiment()` is synchronous, so its hard to imagine a race condition occurring.

Reply:
discussed. we decided to return the tracking call locally in the private _run() method without affecting public run()